modules/dnn/src/model.cpp

```diff
@@ -401,4 +401,242 @@ void DetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
         CV_Error(Error::StsNotImplemented, "Unknown output layer type: \"" + lastLayer->type + "\"");
 }
 
+TextRecognitionModel::TextRecognitionModel(const String& model, const String& config)
+    : Model(model, config) {}
+
+TextRecognitionModel::TextRecognitionModel(const Net& network) : Model(network) {}
+
+void TextRecognitionModel::setVocabulary(const std::vector<String> & voc)
+{
+    this->vocabulary.clear();
+    for (uint i = 0; i < voc.size(); i++) {
+        this->vocabulary.push_back(voc.at(i));
+    }
+}
+
+void TextRecognitionModel::recognize(InputArray frame, String decodeType, std::vector<String> & results,
+                                     const std::vector<std::vector<Point>> & roiPolygons)
+{
+    results.clear();
+
+    std::vector<Mat> outs;
+    uint roiSize = roiPolygons.size();
+    if (roiSize == 0) {
+        impl->predict(*this, frame.getMat(), outs);
+        CV_Assert(outs.size() == 1);
+        Mat prediction = outs[0];
+        String result = decode(prediction, decodeType);
+        results.push_back(result);
```

Comment:
please reduce code duplication - create a vector of inputs and process it after

Reply:
I have updated it. Please check whether I get your mind. Thank you.